src/services/formatting/formatting.ts

```diff
@@ -120,7 +120,9 @@ module ts.formatting {
 
     function findOutermostParent(position: number, expectedTokenKind: SyntaxKind, sourceFile: SourceFile): Node {
         var precedingToken = findPrecedingToken(position, sourceFile);
-        if (!precedingToken || precedingToken.kind !== expectedTokenKind) {
+        if (!precedingToken ||
+            precedingToken.kind !== expectedTokenKind ||
+            position !== precedingToken.getEnd()) {

```

Comment:
Perhaps you could leave a comment for what each of these cases means. I don't see how this corresponds to being in a comment.